src/main/java/duke/command/ByCommand.java

```diff
@@ -0,0 +1,49 @@
+package duke.command;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.exceptions.InvalidTimeFormatException;
+import duke.task.Task;
+import duke.util.DukeIo;
+import duke.util.ParsedData;
+import duke.util.Parser;
+import duke.util.Storage;
+import duke.util.TaskList;
+
+public class ByCommand extends DataCommand {
+
+    /**
+     * Takes in ParsedData potentially containing a datetime pattern.
+     * 
+     * @param d ParsedData containing a possible datetime pattern
+     */
+    public ByCommand(ParsedData d) {
+        super(d);
```

Comment:
Perhaps a more descriptive name rather than "d" is better?

Reply:
changed d -> parsedData.
Although for under the scope of the constructor which is really short, I think this is ok 

>  Variables with a large scope should have long names, variables with a small scope can have short names.